packages/block-editor/src/components/link-control/index.js

```diff
@@ -251,59 +369,121 @@ function LinkControl( {
 			}
 		);
 
-		const manualLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
-		const searchResultsLabelId = isInitialSuggestions
-			? `block-editor-link-control-search-results-label-${ instanceId }`
-			: undefined;
-		const labelText = isInitialSuggestions
-			? __( 'Recently updated' )
-			: sprintf( __( 'Search results for %s' ), inputValue );
+		const directLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+		const isSingleDirectEntryResult =
+			suggestions.length === 1 &&
+			directLinkEntryTypes.includes(
+				suggestions[ 0 ].type.toLowerCase()
+			);
+		const shouldShowCreateEntity =
+			createSuggestion &&
+			! isSingleDirectEntryResult &&
+			! isInitialSuggestions;
+
 		// According to guidelines aria-label should be added if the label
 		// itself is not visible.
 		// See: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role
-		const ariaLabel = isInitialSuggestions ? undefined : labelText;
-		const SearchResultsLabel = (
+		const searchResultsLabelId = `block-editor-link-control-search-results-label-${ instanceId }`;
+		const labelText = isInitialSuggestions
+			? __( 'Recently updated' )
+			: sprintf( __( 'Search results for "%s"' ), inputValue );
+
+		const searchResultsLabel = (
 			<span
-				className="block-editor-link-control__search-results-label"
+				className={ classnames(
+					'block-editor-link-control__search-results-label',
+					{
+						'screen-reader-text': ! isInitialSuggestions,
```

Comment:
Originally this pull request used `VisuallyHidden`, which I noted to be an improvement, because it avoids fragile references to these internal class names. While I can see that it may have been complicated to continue using it _conditionally_, it would still have been possible.

Normally I could imagine that as:

```jsx
createElement( isInitialSuggestions ? 'span' : VisuallyHidden, {
    className: 'block-editor-link-control__search-results-label',
    id: searchResultsLabelId,
}, labelText );
```

One problem is that `VisuallyHidden` doesn't support custom class names. It should. But it doesn't. So you would need to either fork the logic entirely or apply the classname as `components-visually-hidden block-editor-link-control__search-results-label`.

Aside: We should appreciate these moments of pain as illustrations that we should want to avoid complex logic whenever possible, and feel open to use them as justification in pushing back against whatever requirements we think we have. In this case, it would be ideal if we didn't have any conditions and we render consistently no matter whether there's suggestions or not.

Reply:
I took a while considering this. I've decided that it's correct that `VisuallyHidden` should not encourage additional `classNames`.  The styling for VH is very specific and even a slight tweak via custom styles/classes could have unintended a11y consequences.

Therefore I've taken an approach of using it as a wrapping element. It's now conditionally swapped out with a `Fragment` depending on whether we want to hide or not.